### PR TITLE
Add attachment type

### PR DESCRIPTION
**Pull request by Chiggy-Playz**
## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/converter.py [hunk 1]
```diff
@@ -869,6 +869,20 @@ async def convert(self, ctx: Context, argument: str) -> discord.GuildSticker:
...
+class AttachmentConverter(Converter[discord.Attachment]):
+    """Converts to a :class:`~discord.Attachment`.
+
+    This is done by looping through ctx.message attachments to find the attachment with the id
+
+    .. versionadded:: 2.0
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> discord.Attachment:
+        attachment_id = int(argument)
```
Comment (GnomedDev) [inline discord/ext/commands/converter.py:881]: Should a check be done here for if it's an int?

File: discord/ext/commands/converter.py [hunk 2]
```diff
@@ -869,6 +869,20 @@ async def convert(self, ctx: Context, argument: str) -> discord.GuildSticker:
...
+        attachment = [attachment for attachment in ctx.message.attachments if attachment.id == attachment_id][0]
```
Comment (GnomedDev) [inline discord/ext/commands/converter.py:882]: Use `next` and a generator expression

File: discord/ext/commands/converter.py [hunk 3]
```diff
@@ -869,6 +869,20 @@ async def convert(self, ctx: Context, argument: str) -> discord.GuildSticker:
...
+        return attachment
```
Comment (GnomedDev) [inline discord/ext/commands/converter.py:883]: Why are you returning the variable you just created, sourcery would be annoyed at you for this

File: discord/ext/commands/bot.py [hunk 4]
```diff
@@ -1325,6 +1325,16 @@ async def process_slash_commands(self, interaction: discord.Interaction):
                 # String with space in without "consume rest"
                 option = cast(_ApplicationCommandInteractionDataOptionString, option)
                 message.content += f" {_quote_string_safe(option['value'])}"
+            elif option["type"] == 11:
+                message.content += f' {option["value"]}'
+                message.attachments.append(
+                    discord.Attachment(
+                        data=interaction.data["resolved"]["attachments"][
+                            option["value"]
+                        ],  # no clue how to solve this error
```
Comment (GnomedDev) [inline discord/ext/commands/bot.py:1334]: I don't know how black find this code readable but it should be fixed formatting wise

**Review by GnomedDev — CHANGES_REQUESTED** [review]

**Comment by Chiggy-Playz** [comment]
Will be added to class based implementation only.

**Pull request closed without merging**